pkg/karmadactl/promote/promote.go

```diff
@@ -114,6 +114,14 @@ type CommandPromoteOption struct {
 	// DryRun tells if run the command in dry-run mode, without making any server requests.
 	DryRun bool
 
+	// AutoGeneratePropagationPolicy tells if the promote command create a PropagationPolicy automatically, otherwise one can create their own PropagationPolicy
+	AutoGeneratePropagationPolicy bool
```

Comment:
`AutoCreatePropagationPolicy` is better?

Reply:
Sure, will do